locations/spiders/rural_king_us.py

```diff
@@ -0,0 +1,316 @@
+            }
+
+            # Extract email if available
+            for attribute in store.get("Attribute", []):
+                if attribute.get("name") == "Email":
+                    properties["email"] = attribute.get("value")
+                    break
+
+            # Get store image if available
+            for description in store.get("Description", []):
+                if "thumbnail" in description:
+                    properties["image"] = description.get("thumbnail")
+                    break
+
+            # Add hours
+            hours = self.parse_hours(store)
+            if hours:
+                properties["opening_hours"] = hours
+
+            # Create the feature
+            item = Feature(**properties)
+
+            # Apply shop category
+            apply_category(Categories.SHOP_AGRARIAN, item)
+
+            # Add services
+            extras = item.get("extras", {})
+            extras["store_services"] = services
+            item["extras"] = extras
+
+            return item
+
+        except Exception as e:
+            self.logger.error(f"Error parsing store data: {str(e)}")
+            return None
+
+    def parse_hours(self, store):
+        oh = OpeningHours()
+
+        days = {"Mon": "Mo", "Tue": "Tu", "Wed": "We", "Thu": "Th", "Fri": "Fr", "Sat": "Sa", "Sun": "Su"}
+
+        for attribute in store.get("Attribute", []):
+            day_name = attribute.get("displayName")
+            if day_name in days and attribute.get("name", "").startswith("Hours_"):
+                hours_value = attribute.get("value")
+                if hours_value and hours_value.strip():
+                    try:
+                        open_time, close_time = hours_value.split(" - ")
+                        oh.add_range(
+                            day=days[day_name], open_time=open_time, close_time=close_time, time_format="%I:%M%p"
+                        )
+                    except Exception as e:
+                        self.logger.warning(f"Error parsing hours {hours_value} for {day_name}: {e}")
+
+        return oh.as_opening_hours()
+
+
+class RuralKingUSSpider(scrapy.Spider):
+    name = "rural_king_us"
+    item_attributes = {
+        "brand": "Rural King",
+        "brand_wikidata": "Q7380525",
+    }
+    allowed_domains = ["ruralking.com"]
+    download_delay = 1.0  # Be nice to their servers
+
+    def start_requests(self):
+        # Check if a specific URL was passed for testing
+        if hasattr(self, "url") and self.url:
+            yield scrapy.Request(self.url, callback=self.parse_api)
+        else:
+            # Use searchable points to query for stores
+            # This approach allows for complete coverage across the US
+            # and accommodates future expansion to new regions
+
+            with open("./locations/searchable_points/us_centroids_50mile_radius.csv") as points_file:
+                next(points_file)  # Skip the header
```

Comment:
Instead of iterating over the CSV yourself, you can use [`point_locations`](https://github.com/alltheplaces/alltheplaces/blob/master/locations/geo.py#L41C5-L41C30) or [`country_seadgg_centroids`](https://github.com/alltheplaces/alltheplaces/blob/master/locations/geo.py#L41C5-L41C30)